x509-cert/src/ext/pkix/keyusage.rs

```diff
@@ -136,6 +137,20 @@ impl AssociatedOid for ExtendedKeyUsage {
 }
 
 impl_newtype!(ExtendedKeyUsage, Vec<ObjectIdentifier>);
+// TODO
+//   This extension MAY, at the option of the certificate issuer, be
+//   either critical or non-critical.
+//
+//   If a CA includes extended key usages to satisfy such applications,
+//   but does not wish to restrict usages of the key, the CA can include
+//   the special KeyPurposeId anyExtendedKeyUsage in addition to the
+//   particular key purposes required by the applications.  Conforming CAs
+//   SHOULD NOT mark this extension as critical if the anyExtendedKeyUsage
+//   KeyPurposeId is present.  Applications that require the presence of a
+//   particular purpose MAY reject certificates that include the
+//   anyExtendedKeyUsage OID but not the particular OID expected for the
+//   application.
```

Comment:
This makes me consider rewrite the `AsExtension` trait as the `critical` depends on other extensions already provided.